applications/crossbar/src/api_resource.erl

```diff
@@ -796,6 +797,29 @@ to_json(Req, Context, Accept) ->
             (?MODULE):Fun(Req, Context)
     end.
 
+-spec to_xml(cowboy_req:req(), cb_context:context()) -> {iolist(), cowboy_req:req(), cb_context:context()}.
+to_xml(Req, Context) ->
+    lager:debug("run: to_xml"),
+    [{Mod, _Params}|_] = cb_context:req_nouns(Context),
+    Verb = cb_context:req_verb(Context),
+    Event = api_util:create_event_name(Context, [<<"to_xml">>
+                                                ,kz_term:to_lower_binary(Verb)
+                                                ,Mod
+                                                ]),
+    {Req1, Context1} = crossbar_bindings:fold(Event, {Req, Context}),
+    case cb_context:fetch(Context1, 'is_chunked') of
```

Comment:
This won't work anymore. In 4.2+ we added a generic chunked response to crossbar. You can look at `to_json/2` or `to_csv/2` as example.

Basically you have to use `crossbar_view` to get view results,. If the request wants chunked response `crossbar_view` stores the necessary values in `Context` and then in `api_resource:to_xml/2`, if `is_chunk` is set, it should call `to_chunk/3` which then starts getting results from view and send response to client in chunk.

You simply have to write appropriate function clause for `send_chunk_response/3` and `finish_chunked_response/1` to generate XML from JSON.